Add command to remove orphan bundles

castulo opened the pull request:
Dependencies of orphaned bundles can cause bloat and unexpected content to be on disk.
We need to remove all bundles that are not tracked or dependency of a tracked bundle.
This option should also be available for 3rd-party.

Closes #1529

castulo commented:
@otaviobp I forgot to mention this, but the PR is ready for review regardless of the failing job. That is again because of new tests in the API tests.

otaviobp commented on src/3rd_party_bundle_remove.c:
```diff
@@ -164,26 +190,33 @@ enum swupd_code third_party_bundle_remove_main(int argc, char **argv)
 	/* set the command options */
 	bundle_remove_set_option_force(cmdline_option_force);
 	bundle_remove_set_option_recursive(cmdline_option_recursive);
+	bundle_remove_set_option_orphans(cmdline_option_orphans);
```
Parameters are better than setting a static variable. It's out of the scope of this PR, but both cmdline_option_force and cmdline_option_recursive could become parameters of execute_remove_bundles_extra

castulo commented on src/bundle_remove.c:
```diff
@@ -345,10 +386,18 @@ enum swupd_code execute_remove_bundles_extra(struct list *bundles, remove_extra_
 	}
 
-	for (; bundles; bundles = bundles->next, total++) {
+	if (cmdline_option_orphans) {
```
I can do that, but it has its disadvantages that is why I didn't take this route, for example we will have to load the manifests outside a process step, it also requires more code duplication.

castulo commented on src/bundle_remove.c:
```diff
@@ -271,18 +302,28 @@ static void print_remove_summary(unsigned int requested, unsigned int bad, unsig
 
-	if (bad > 0) {
-		info("\nFailed to remove %i of %i bundles\n", bad, requested);
+	if (cmdline_option_orphans) {
```
what should we show when the user runs bundle-remove --orphans and nothing gets removed because there are no orphan bundles?

castulo commented:
V2:
- Rebased
- Used a new pointer to iterate over the bundles list
- Removed the word "orphaned" from the bundle-remove summary so instead of saying: "Successfully removed 1 orphaned bundle" it will say "Successfully removed 1 bundle"

castulo commented:
I didn't change the logic in the code to pass the list of orphaned bundles to `execute_remove_bundles_extra()` because this IMO has bigger disadvantages, like having to load the manifests outside a process step, and requiring more code duplication.

castulo commented:
I forgot to push changes for a couple of tests after pushing V2. It should be good now.
